xmr-btc/tests/harness/storage.rs

```diff
@@ -0,0 +1,159 @@
+use anyhow::{anyhow, Context, Result};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::path::Path;
+
+pub struct Database {
+    db: sled::Db,
+}
+
+impl Database {
+    const LAST_STATE_KEY: &'static str = "latest_state";
+
+    pub fn open(path: &Path) -> Result<Self> {
+        let path = path
+            .to_str()
+            .ok_or_else(|| anyhow!("The path is not utf-8 valid: {:?}", path))?;
```

Comment:
No need to convert path to a string, `open` takes `AsRef<Path>`.
```
pub fn open<P: AsRef<Path>>(path: P) -> Result<Db>
```

Reply:
good pick